Build issue on expo EAS

```
> Task :qonversion_react-native-sdk:compileDebugKotlin
w: file:///home/expo/workingdir/build/node_modules/@qonversion/react-native-sdk/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt:26:47 'val currentActivity: Activity?' is deprecated. Deprecated in 0.80.0. Use getReactApplicationContext().getCurrentActivity() instead.
w: file:///home/expo/workingdir/build/node_modules/@qonversion/react-native-sdk/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt:26:47 This synthetic property is based on the getter function 'fun getCurrentActivity(): Activity?' from Kotlin. In the future, synthetic properties will be available only if the base getter function came from Java. Consider replacing this property access with a 'getCurrentActivity()' function call.
w: file:///home/expo/workingdir/build/node_modules/@qonversion/react-native-sdk/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt:302:46 The corresponding parameter in the supertype 'QonversionEventsListener' is named 'transaction'. This may cause problems when calling this function with named arguments.
```

Please fix deprecated code paths

Hi, thanks for the report!                                                 
   
The lines you pasted are Kotlin compiler warnings (`w:`), not errors - they shouldn't fail the Gradle build on their own. Could you share the full EAS build log, specifically anything around `BUILD FAILED` or an `e:` (error) line? 
That will tell us whether the build is actually failing or just noisy.

Also helpful:
- React Native version
- Expo SDK version
- Your EAS build profile (any `allWarningsAsErrors` / `-Werror` flags?)
                                                                                
Independently, the deprecation warnings are valid, and we'll clean them up in an upcoming release.

--- a/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt
+++ b/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt
@@ -299,7 +299,6 @@ class QonversionModule(reactContext: ReactApplicationContext) : NativeQonversion
         emitOnEntitlementsUpdated(mappedEntitlements)
     }
 
-    @Suppress("PARAMETER_NAME_CHANGED_ON_OVERRIDE")
     override fun onDeferredPurchaseCompleted(purchaseResult: BridgeData) {
         val mappedPurchaseResult = EntitiesConverter.convertMapToWritableMap(purchaseResult)
         emitOnDeferredPurchaseCompleted(mappedPurchaseResult)

--- a/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt
+++ b/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt
@@ -23,7 +23,7 @@ class QonversionModule(reactContext: ReactApplicationContext) : NativeQonversion
         (reactContext.applicationContext as Application),
         object : ActivityProvider {
             override val currentActivity: Activity?
-                get() = this@QonversionModule.currentActivity
+                get() = reactApplicationContext.currentActivity
         },
         this
     )
@@ -299,9 +299,9 @@ class QonversionModule(reactContext: ReactApplicationContext) : NativeQonversion
         emitOnEntitlementsUpdated(mappedEntitlements)
     }
 
-    override fun onDeferredPurchaseCompleted(purchaseResult: BridgeData) {
-        val mappedPurchaseResult = EntitiesConverter.convertMapToWritableMap(purchaseResult)
-        emitOnDeferredPurchaseCompleted(mappedPurchaseResult)
+    override fun onDeferredPurchaseCompleted(transaction: BridgeData) {
+        val mappedTransaction = EntitiesConverter.convertMapToWritableMap(transaction)
+        emitOnDeferredPurchaseCompleted(mappedTransaction)
     }
 
     companion object {

--- a/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt
+++ b/android/src/main/java/com/qonversion/reactnativesdk/QonversionModule.kt
@@ -299,9 +299,10 @@ class QonversionModule(reactContext: ReactApplicationContext) : NativeQonversion
         emitOnEntitlementsUpdated(mappedEntitlements)
     }
 
-    override fun onDeferredPurchaseCompleted(transaction: BridgeData) {
-        val mappedTransaction = EntitiesConverter.convertMapToWritableMap(transaction)
-        emitOnDeferredPurchaseCompleted(mappedTransaction)
+    @Suppress("PARAMETER_NAME_CHANGED_ON_OVERRIDE")
+    override fun onDeferredPurchaseCompleted(purchaseResult: BridgeData) {
+        val mappedPurchaseResult = EntitiesConverter.convertMapToWritableMap(purchaseResult)
+        emitOnDeferredPurchaseCompleted(mappedPurchaseResult)
     }
 
     companion object {

